jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -185,6 +185,13 @@ def _on_link_type_updated(self, msg=None):
         # toggle visibility as necessary
         self._on_is_active_changed()
 
+        # When fp(s) added via API before WCS link. Overlays visibility & is_active
+        # can be True, but only last fp will display. This ensures all fp(s) display
+        if not self.is_pixel_linked:
+            for choice in self.overlay.choices:
+                if len(self.overlay.choices) > 1:
+                    self._change_overlay(overlay_selected=choice)

```

Comment:
if there aren't any choices, then the for-loop shouldn't even enter.  It might also be a good idea to add a comment here so we don't think this is changing the selection (feel free to edit the text to whatever you think makes sense).

```suggestion
            for choice in self.overlay.choices:
                # trigger the update without actually changing the user-selection
                self._change_overlay(overlay_selected=choice)

```

Reply:
When Imviz is instantiated in a notebook or test, the `default` overlay is temporarily in `self.overlay.choice`(and it is NoneType at this time). So to make sure that a TypeError exception isn't thrown I added this conditional.

I will add a comment now for some clarity on what this loop is doing.

Reply:
so it enters the loop and then the choice disappears before `_change_overlay` is called?

Reply:
The default does get passed into `_change_overlay`, it does go through creating a new entry, at the end of creating a new entry, since there is now 1 entry, the overlay tries to `center_on_viewer`. And when this happens the `viewer.get_center_skycoord` is what is actually NoneType (I mistakenly said it was the overlay). 

In the case that no overlay is added in the API before WCS linking, the default that gets created does still center in the viewer (so this does still trigger correctly if there is only 1 overlay).